src/test/java/uk/gov/hmcts/reform/laubackend/idam/repository/UserDeletionAuditRepositoryTest.java

```diff
@@ -265,4 +268,59 @@ static DeletionLogGetRequestParams getRequestParams(
     static Pageable getPage() {
         return PageRequest.of(0, 100);
     }
+
+    @Test
+    void shouldFindPageableResultsForAllUsers() {
+        final DeletionLogAllUsersRequestParams params = new DeletionLogAllUsersRequestParams(
+            "10",
+            "1",
+            "asc"
+        );
+        final Page<UserDeletionAudit> userDeletion = userDeletionAuditFindRepository.findAllDeletedUsers(
+            ENCRYPTION_KEY,
+            PageRequest.of(Integer.parseInt(params.page()) - 1, Integer.parseInt(params.size()),
+                           isEmpty(params.sort()) ? Sort.Direction.DESC : Sort.Direction.fromString(params.sort()),
+                           "deletion_timestamp")
+        );
+
+        assertThat(userDeletion.getTotalElements()).isEqualTo(10);
+        assertThat(userDeletion.getContent()).hasSize(10);
+    }
+
+    @Test
+    void shouldFindResultsForAllUsers() {
+        final DeletionLogAllUsersRequestParams params = new DeletionLogAllUsersRequestParams(
+            "10000",
+            "1",
+            "desc"
+        );
+        final Page<UserDeletionAudit> userDeletion = userDeletionAuditFindRepository.findAllDeletedUsers(
+            ENCRYPTION_KEY,
+            PageRequest.of(Integer.parseInt(params.page()) - 1, Integer.parseInt(params.size()),
+                           isEmpty(params.sort()) ? Sort.Direction.DESC : Sort.Direction.fromString(params.sort()),
+                           "deletion_timestamp")
+        );
+
+        assertThat(userDeletion.getTotalElements()).isEqualTo(20);
+        assertThat(userDeletion.getContent()).hasSize(20);
+    }
+
+    @Test
+    void shouldFindResultsForAllUsersWithoutSort() {
```

Comment:
We are not asserting the Sort direction, please verify that the records are sorted by desc order

Reply:
Records direction is now getting tested

Reply:
The last line where I 'm getting content , it is fetching 0th content and checking if its is 20th when descending and 1 when ask

Reply:
Please see assertEquals method